### PR TITLE
Logging preferences

### PR DESCRIPTION
**Pull request by ineiti**
More useful logging done while searching for additional bucket creation.

### CODE REVIEW

File: .github/workflows/lint_build_test.yaml [hunk 1]
```diff
@@ -106,7 +106,13 @@ jobs:
...
 
       - name: running tests
-        run: go list -f '{{.Dir}}' ./... | xargs -n1 go test -race -short
+        run: |
+          for d in $( go list -f '{{.Dir}}' ./... ); do
+            echo "Testing $d"
+            if ! go test -race -short -v $d > log.txt; then
+              cat log.txt
+            fi
+          done
```
[the remark this answers is not in the document]
Comment (ineiti) [reply .github/workflows/lint_build_test.yaml:115]: I'm not sure we're talking about the same thing. If I run `go test -v`, it shows me all the logs up to verbose level 2, whether it is successful, or not. `go test` doesn't show any logs other than `log.Info`.
So this is supposed to show the logs up to verbose level 2, but only if it fails.

[the remark this answers is not in the document]
Comment (ineiti) [reply .github/workflows/lint_build_test.yaml:115]: @tharvik says that if DEBUG_LVL=2, it should also work.

[the remark this answers is not in the document]
Comment (ineiti) [reply .github/workflows/lint_build_test.yaml:115]: Did I understand correctly that you expect these two calls to produce the same output? I modified the `NewLedgerCorrupted` test to produce an error...

Just using `DEBUG_LVL=2`
```
$ DEBUG_LVL=2 go test -run Corrupted
--- FAIL: TestClient_NewLedgerCorrupted (1.72s)
    api_test.go:39: 
        	Error Trace:	api_test.go:39
        	Error:      	Should be true
        	Test:       	TestClient_NewLedgerCorrupted
FAIL
exit status 1
FAIL	go.dedis.ch/cothority/v3/byzcoin	2.443s
```

Using `-v`
```
$ go test -run NewLedgerCorrupted -v
=== RUN   TestClient_NewLedgerCorrupted
1 21/04/01 13:52:17.379804000: server.go:217 (v3.(*Server).Start)       - Starting server at 2021-01-04 13:52:17 on address tcp://127.0.0.1:60389
2 21/04/01 13:52:17.379905000: websocket.go:182 (v3.(*WebSocket).start) - Starting to listen on 0.0.0.0:60390
1 21/04/01 13:52:17.971244000: server.go:217 (v3.(*Server).Start)       - Starting server at 2021-01-04 13:52:17 on address tcp://127.0.0.1:60395
2 21/04/01 13:52:17.971299000: websocket.go:182 (v3.(*WebSocket).start) - Starting to listen on 0.0.0.0:60396
1 21/04/01 13:52:18.602324000: server.go:217 (v3.(*Server).Start)       - Starting server at 2021-01-04 13:52:18 on address tcp://127.0.0.1:60397
2 21/04/01 13:52:18.602371000: websocket.go:182 (v3.(*WebSocket).start) - Starting to listen on 0.0.0.0:60398
    TestClient_NewLedgerCorrupted: api_test.go:39: 
        	Error Trace:	api_test.go:39
        	Error:      	Should be true
        	Test:       	TestClient_NewLedgerCorrupted
1 21/04/01 13:52:18.656088000: service.go:2644 (byzcoin.(*Service).cleanupGoroutines) - tcp://127.0.0.1:60389 closing go-routines
1 21/04/01 13:52:18.656128000: service.go:2644 (byzcoin.(*Service).cleanupGoroutines) - tcp://127.0.0.1:60397 closing go-routines
1 21/04/01 13:52:18.656142000: service.go:2644 (byzcoin.(*Service).cleanupGoroutines) - tcp://127.0.0.1:60395 closing go-routines
--- FAIL: TestClient_NewLedgerCorrupted (1.84s)
FAIL
exit status 1
FAIL	go.dedis.ch/cothority/v3/byzcoin	2.025s
```

[the remark this answers is not in the document]
Comment (tharvik) [reply .github/workflows/lint_build_test.yaml:115]: Yes, it is what I meant, and it is working on my side.

```
tharvik@t480s cothority % go test -run TestError_ErrorOrNil
--- FAIL: TestError_ErrorOrNil (0.00s)
    error_test.go:22:
TestError_ErrorOrNilSkip
FAIL
exit status 1
FAIL	go.dedis.ch/cothority/v3	0.021s
```
```
tharvik@t480s cothority % DEBUG_LVL=2 go test -run TestError_ErrorOrNil
2 : error_test.go:21 (v3.TestError_ErrorOrNil) - info
--- FAIL: TestError_ErrorOrNil (0.00s)
    error_test.go:22:
TestError_ErrorOrNilSkip
FAIL
exit status 1
FAIL	go.dedis.ch/cothority/v3	0.020s
```

In your specific case, `byzcoin/service_test.go:TestMain` is calling `log.MainTest` which reset the `DEBUG_LVL` to some other value thus not showing anything. IMO, you can remove it.

[the remark this answers is not in the document]
Comment (ineiti) [reply .github/workflows/lint_build_test.yaml:115]: Hmm - `log.MainTest` also does check that once the test is over no go-routines are hanging around... So I'll have to look how I can put the two together.

**Comment by ineiti** [comment]
Replaced by dedis/onet#673

**Pull request closed without merging**